packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -59,6 +53,53 @@ interface TableRowExpansionToggleEventDetail {
 
 Note: This event will _not_ be emitted for group rows.
 
+### `Table` API for specifying lazy loaded hierarchical data
+
+In some cases, a client might know that a given row has children, but for performance reasons, that data hasn't been loaded from the backend and given to the table. To enable this use case, there will be an API on the table to specify that a given row should be expandable even if it has no children within the data and to specify that a row is loading its children. That API will look as follows:
+
+```ts
+interface TableRowOptions {
+    // Specifies whether or not the row should show the expand-collapse button
+    // even if it does not have any children in the data.
+    forceExpandable: boolean;
+
+    // Specifies whether or not the row should be in a state of loading its
+    // children.
+    loading: boolean;
+}
+
+public Table() {
+    ...
+    // Sets the options for the rows specified by the passed IDs.
+    public async setRowOptions(rowOptions: { id: string, options: TableRowOptions }[]): Promise<void>;
+}
+```
+
+The loading state of a row will look as follows:
+
+![Lazy Loading Spinner](./spec-images/LazyLoadingSpinner.gif)
+
+Some notes about the `setRowOptions` API:
+
+-   previously set row options will be cleared when `setData` is called
+-   previously set row options will be cleared when the table's `idFieldName` changes
```

Comment:
Good question, and the answer actually depends on what type of information we think will ultimately be in the options object for each row. Therefore, I've updated the direction so that `setRowOptions` is now `setHierarchyRowOptions`, with the intention that those options will only be related to hierarchy. In which case, yes, we should clear those options if the `parentIdFieldName` changes.

Reply:
After a bit of discussion with @rajsite and @atmgrifter00, I've changed this to no longer clear the options when `parentIdFieldName` changes. I've made some other behavior changes around `setRowHierarchyOptions` (including renaming it to `setRecordHierarchyOptions`), and one of those changes now makes it easy for a client to clear themselves by passing an empty array of options. Therefore, if they are explicitly changing the hierarchy settings of a table in their application, they can also explicitly clear the hierarchy options if the previously set ones don't make sense for their new configuration.